prow/github/client.go

```diff
@@ -369,23 +369,11 @@ func (c *Client) requestRetry(method, path, accept string, body interface{}) (*h
 				if resp.Header.Get("X-RateLimit-Remaining") == "0" {
 					// If we are out of API tokens, sleep first. The X-RateLimit-Reset
 					// header tells us the time at which we can request again.
-					var t int
-					if t, err = strconv.Atoi(resp.Header.Get("X-RateLimit-Reset")); err == nil {
-						// Sleep an extra second plus how long GitHub wants us to
-						// sleep. If it's going to take too long, then break.
-						sleepTime := c.time.Until(time.Unix(int64(t), 0)) + time.Second
-						if sleepTime < maxSleepTime {
-							c.time.Sleep(sleepTime)
-						} else {
-							err = fmt.Errorf("sleep time for token reset exceeds max sleep time (%v > %v)", sleepTime, maxSleepTime)
-							resp.Body.Close()
-							break
-						}
-					} else {
-						err = fmt.Errorf("failed to parse rate limit reset unix time %q: %v", resp.Header.Get("X-RateLimit-Reset"), err)
-						resp.Body.Close()
-						break
-					}
+					c.waitOrClose("X-RateLimit-Reset", resp)
```

Comment:
We still need to `break` from the retry loop if we errored and closed the response body.

Reply:
Of course, sorry -- was hurried. Fixed

Reply:
Now we are always `break`ing. We should only `break` if we errored and closed the response body. If we slept we need to proceed to the next iteration of the retry loop.